checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -133,166 +411,116 @@ protected boolean checkOverride(MethodTree overriderTree,
-                TypeMirror rt = methodElement.getReceiverType();
-
-                if (rt != null) {
-                    List<? extends AnnotationMirror> amList = rt.getAnnotationMirrors();
-
-                    for (AnnotationMirror annotationMirror : amList) {
-
-                        if (AnnotationUtils.areSameByClass( annotationMirror, checkerGuardedByClass) ||
-                            AnnotationUtils.areSameByClass( annotationMirror, javaxGuardedByClass) ||
-                            AnnotationUtils.areSameByClass( annotationMirror, jcipGuardedByClass)) {
-                            List<String> guardedByValue = AnnotationUtils.getElementValueArray(annotationMirror, "value", String.class, false);
+        if (!isMethodCall) {
+            ExpressionTree expr = null;
+
+            switch(treeKind) {
+                case ARRAY_ACCESS:
+                    expr = ((ArrayAccessTree) tree).getExpression();
+                    break;
+                case MEMBER_SELECT:
+                    expr = ((MemberSelectTree) tree).getExpression();
+                    break;
+                default:
+                    expr = tree;
+                    break;
+            }
 
-                            // A GuardedBy annotation on the receiver of the enclosing method was found.
-                            // Now check if the receiver on the enclosing method is the same as the receiver
-                            // on the field/method we are checking preconditions for.
+            Element invokedElement = TreeUtils.elementFromUse(expr);
 
-                            if (guardedByValue != null) {
-                                Node nodeNode = atypeFactory.getNodeForTree(tree);
-                                Node receiverNode = null;
+            AnnotatedTypeMirror atmOfReceiverOrIdentifier = atypeFactory.getReceiverType(tree);
 
-                                if (nodeNode instanceof FieldAccessNode) {
-                                    receiverNode = ((FieldAccessNode) nodeNode).getReceiver();
-                                }
-                                else if (nodeNode instanceof MethodInvocationNode)
-                                {
-                                    MethodAccessNode man = ((MethodInvocationNode) nodeNode).getTarget();
-
-                                    if (!man.getMethod().getModifiers().contains(Modifier.STATIC)) {
-                                        receiverNode = man.getReceiver();
-                                    }
-                                }
+            Node node = atypeFactory.getNodeForTree(tree);
 
-                                if (receiverNode instanceof ExplicitThisLiteralNode ||
-                                    receiverNode instanceof ImplicitThisLiteralNode) {
+            // Is the receiver of the expression being accessed the same as the receiver of the enclosing method?
+            boolean receiverIsThatOfEnclosingMethod = false;
```

Comment:
I would move this declaration and the code that calculates the correct value close to it use.  Otherwise, a reader has to scroll back to this point to figure out how (and why) it is calculated.